File: TwoWayDatabinding/app/src/main/java/com/enpassio/twowaydatabinding/viewmodel/AddToyViewModelFactory.java
package com.enpassio.twowaydatabinding.viewmodel;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.enpassio.twowaydatabinding.data.ToyRepository;
import com.enpassio.twowaydatabinding.data.model.ToyEntry;

public class AddToyViewModelFactory extends ViewModelProvider.NewInstanceFactory {

    private ToyRepository mRepo;
    private ToyEntry mChosenToy;

    public AddToyViewModelFactory(ToyRepository repo, ToyEntry chosenToy) {
        mRepo = repo;
        mChosenToy = chosenToy;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        //noinspection unchecked
        return (T) new AddToyViewModel(mRepo, mChosenToy);
    }
}

File: TwoWayDatabindingKotlin/app/src/main/java/com/enpassion/twowaydatabindingkotlin/viewmodel/AddToyViewModelFactory.kt
package com.enpassion.twowaydatabindingkotlin.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.enpassion.twowaydatabindingkotlin.data.ToyEntry
import com.enpassion.twowaydatabindingkotlin.data.ToyRepository


class AddToyViewModelFactory(private val mRepo: ToyRepository, private val mChosenToy: ToyEntry?) :
    ViewModelProvider.NewInstanceFactory() {

    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return AddToyViewModel(mRepo, mChosenToy) as T
    }
}